cmd/provisioner-localpv/app/helper.go

```diff
@@ -0,0 +1,177 @@
+	}
+	path = hostPath.Path
+
+	nodeAffinity := pv.Spec.NodeAffinity
+	if nodeAffinity == nil {
+		return "", "", fmt.Errorf("no NodeAffinity set")
+	}
+	required := nodeAffinity.Required
+	if required == nil {
+		return "", "", fmt.Errorf("no NodeAffinity.Required set")
+	}
+
+	node = ""
+	for _, selectorTerm := range required.NodeSelectorTerms {
+		for _, expression := range selectorTerm.MatchExpressions {
+			if expression.Key == KeyNode && expression.Operator == v1.NodeSelectorOpIn {
+				if len(expression.Values) != 1 {
+					return "", "", fmt.Errorf("multiple values for the node affinity")
+				}
+				node = expression.Values[0]
+				break
+			}
+		}
+		if node != "" {
+			break
+		}
+	}
+	if node == "" {
+		return "", "", fmt.Errorf("cannot find affinited node")
+	}
+	return path, node, nil
+}
+
+// createCleanupPod launches a helper(busybox) pod, to delete the host path.
+//  This porivsioner expects that the host paths are created using
+//  an unique PV path - under a given BasePath. From the absolute path,
+//  it extracs the base path and the PV path. The helper pod is then launched
+//  by mounting the base path - and performing a delete on the unique PV path.
+func (p *Provisioner) createCleanupPod(cmdsForPath []string, name, path, node string) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "failed to cleanup volume %v", name)
+	}()
+	if name == "" || path == "" || node == "" {
+		return fmt.Errorf("invalid empty name or path or node")
+	}
+
+	//Validate that non-root directories are not passed for delete
+	// and also perform white/black list validations.
+	config := &CASConfigPVC{}
+	path, err = config.validatePath(path)
+	if err != nil {
+		return err
+	}
+
+	// Extract the base path and the volume unique path.
+	path = strings.TrimSuffix(path, "/")
+	parentDir, volumeDir := config.extractSubPath(path)
+	//parentDir, volumeDir := filepath.Split(path)
+	//parentDir = strings.TrimSuffix(parentDir, "/")
+	//volumeDir = strings.TrimSuffix(volumeDir, "/")
+
+	//hostPathType := v1.HostPathDirectoryOrCreate
+	//TODO Convert the following into an builder pattern
+	helperPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cleanup-" + name,
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			NodeName:      node,
+			Containers: []v1.Container{
+				{
+					Name:    "local-path-cleanup",
+					Image:   p.helperImage,
+					Command: append(cmdsForPath, filepath.Join("/data/", volumeDir)),
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "data",
+							ReadOnly:  false,
+							MountPath: "/data/",
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: "data",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: parentDir,
+							//Type: &hostPathType,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	//Launch the cleanup pod.
+	pod, err := p.kubeClient.CoreV1().Pods(p.namespace).Create(helperPod)
```

Comment:
Maya has build wrappers for all CRUD operations against any native or custom resource.
This code will look like following:
```go
import (
  pod "github.com/openebs/maya/pkg/kubernetes/pod/v1alpha1"
)

podc, err := pod.KubeClient(pod.WithNamespace(p.namespace)).Create(p)
```

Reply:
https://github.com/openebs/maya/issues/1168